core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -215,16 +216,27 @@ public static FileIO loadFileIO(
       Configuration hadoopConf) {
     LOG.info("Loading custom FileIO implementation: {}", impl);
     DynConstructors.Ctor<FileIO> ctor;
+    boolean useNoArgConstructor = true;
     try {
       ctor = DynConstructors.builder(FileIO.class).impl(impl).buildChecked();
     } catch (NoSuchMethodException e) {
-      throw new IllegalArgumentException(String.format(
-          "Cannot initialize FileIO, missing no-arg constructor: %s", impl), e);
+      LOG.error("No-arg constructor not found for {}, try to use Hadoop Configuration constructor", impl, e);
```

Comment:
try -> will attempt to use

Just because the current wording makes it sound like the user needs to try to do something

Reply:
Also shouldn't this be "info"? It's not necessarily an error right?

Reply:
Thanks, updated